cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaConnectApiImpl.java

```diff
@@ -0,0 +1,478 @@
+    }
+
+    @Override
+    public Future<Map<String, Map<String, String>>> listConnectorLoggers(String host, int port) {
+        Promise<Map<String, Map<String, String>>> result = Promise.promise();
+        HttpClientOptions options = new HttpClientOptions().setLogActivity(true);
+        String path = "/admin/loggers/";
+        vertx.createHttpClient(options)
+                .get(port, host, path, response -> {
+                    response.exceptionHandler(error -> {
+                        result.fail(error);
+                    });
+                    if (response.statusCode() == 200) {
+                        response.bodyHandler(buffer -> {
+                            ObjectMapper mapper = new ObjectMapper();
+
+                            try {
+                                Map<String, Map<String, String>> fetchedLoggers = mapper.readValue(buffer.getBytes(), MAP_OF_MAP_OF_STRINGS);
+                                result.complete(fetchedLoggers);
+                            } catch (IOException e) {
+                                log.warn("Failed to get list of connector loggers", e);
+                                result.fail(new ConnectRestException(response, "Failed to get connector loggers", e));
+                            }
+                        });
+                    } else {
+                        result.fail(new ConnectRestException(response, "Unexpected status code"));
+                    }
+                })
+                .exceptionHandler(result::fail)
+                .setFollowRedirects(true)
+                .putHeader("Accept", "application/json")
+                .end();
+        return result.future();
+    }
+
+    private Future<Void> updateLoggers(String host, int port, String desiredLogging, Map<String, Map<String, String>> fetchedLoggers) {
+        Map<String, String> updateLoggers = new LinkedHashMap<>();
+        fetchedLoggers.entrySet().forEach(entry -> {
+            // set all logger levels to OFF
+            updateLoggers.put(entry.getKey(), "OFF");
+        });
+
+        OrderedProperties ops = new OrderedProperties();
+        ops.addStringPairs(desiredLogging);
+        ops.asMap().entrySet().forEach(entry -> {
+            // set desired loggers to desired levels
+            if (entry.getKey().equals("log4j.rootLogger")) {
+                updateLoggers.put("root", entry.getValue());
+            } else if (!entry.getKey().startsWith("log4j.appender")) {
+                updateLoggers.put(entry.getKey().replace("log4j.logger.", ""), entry.getValue());
+            }
+        });
+
+        LinkedHashMap<String, String> updateSortedLoggers = sortLoggers(updateLoggers);
+        Future<Void> result = Future.succeededFuture();
+        for (Map.Entry<String, String> logger : updateSortedLoggers.entrySet()) {
+            result = result.compose(previous -> updateConnectorLogger(host, port, logger.getKey(), logger.getValue().split(",")[0].replaceAll("\\s", "")));
```

Comment:
I think you should refactor the `.split(",")[0].replaceAll("\\s", "")`  into a method and provide a comment explaining it.